sdk/test-utils/recorder/src/sanitizer.ts

```diff
@@ -1,230 +1,305 @@
+    httpClient: HttpClient,
+    url: string,
+    recordingId: string,
+    sanitizer: Record<string, unknown>
+  ) => {
+    await addSanitizer(httpClient, url, recordingId, {
+      sanitizer: sanitizerName,
+      body: sanitizer,
+    });
+  };
 
-    if (options.removeHeaderSanitizer) {
-      this.addSanitizer({
-        sanitizer: "RemoveHeaderSanitizer",
-        body: JSON.stringify({
-          headersForRemoval: options.removeHeaderSanitizer.headersForRemoval.toString(),
-        }),
+/**
+ * Makes an AddSanitizer<boolean> function that adds the sanitizer if the value is set to true,
+ * and otherwise makes no request to the server.
+ */
+const makeAddBodilessSanitizer =
+  (sanitizerName: ProxyToolSanitizers): AddSanitizer<boolean> =>
+  async (httpClient: HttpClient, url: string, recordingId: string, enable: boolean) => {
+    if (enable) {
+      await addSanitizer(httpClient, url, recordingId, {
+        sanitizer: sanitizerName,
+        body: undefined,
       });
     }
+  };
 
-    if (options.continuationSanitizers) {
-      // TODO: Test
-      await Promise.all(
-        options.continuationSanitizers.map((replacer) =>
-          this.addSanitizer({
-            sanitizer: "ContinuationSanitizer",
-            body: JSON.stringify({
-              ...replacer,
-              resetAfterFirst: replacer.resetAfterFirst.toString(),
-            }),
-          })
-        )
-      );
-    }
-
-    if (options.uriSubscriptionIdSanitizer) {
-      await this.addSanitizer({
-        sanitizer: "UriSubscriptionIdSanitizer",
-        body: JSON.stringify(options.uriSubscriptionIdSanitizer),
+/**
+ * Makes an AddSanitizer function for a FindReplaceSanitizer, for example a bodySanitizer.
+ * Depending on the input FindReplaceSanitizer options, either adds a sanitizer named `regexSanitizerName`
+ * or `stringSanitizerName`.
+ */
+const makeAddFindReplaceSanitizer =
+  (
+    regexSanitizerName: ProxyToolSanitizers,
+    stringSanitizerName: ProxyToolSanitizers
+  ): AddSanitizer<FindReplaceSanitizer> =>
+  async (
+    httpClient: HttpClient,
+    url: string,
+    recordingId: string,
+    sanitizer: FindReplaceSanitizer
+  ): Promise<void> => {
+    if (isStringSanitizer(sanitizer)) {
+      await addSanitizer(httpClient, url, recordingId, {
+        sanitizer: stringSanitizerName,
+        body: {
+          target: sanitizer.target,
+          value: sanitizer.value,
+        },
+      });
+    } else {
+      await addSanitizer(httpClient, url, recordingId, {
+        sanitizer: regexSanitizerName,
+        body: {
+          regex: sanitizer.target,
+          value: sanitizer.value,
+          groupForReplace: sanitizer.groupForReplace,
+        },
       });
     }
+  };
+
+/**
+ *  Internally,
+ * - connection strings are parsed and
+ * - each part of the connection string is mapped with its corresponding fake value
+ * - generalRegexSanitizer is applied for each of the parts with the real and fake values that are parsed
```

Comment:
// general-string-sanitizer is applied fo....

Reply:
Good catch.